### PR TITLE
Better way of getting goal bar info and display in-progress percentage

### PR DESCRIPTION
**Pull request by lmsv-mx123**
Change from approximation of displaying percentage from target level (i.e. each 7 levels being halfway) and also displays in progress percentage, which is that between current level and next level

### CODE REVIEW

**Review by conker-rsc — APPROVED** [review]
I tested in-game. Really cool change, nice idea. Code looks fine

File: src/Game/XPBar.java [hunk 1]
```diff
@@ -156,6 +156,10 @@ void draw(Graphics2D g, BufferedMouseClick bufferedMouseClick) {
       g.setColor(Renderer.color_fatigue);
     }
     g.fillRect(x + 1, y + 1, percent, bounds.height - 2);
+    if (percent_progress > 0.0f) {
+      g.setColor(Renderer.color_fatigue);
+      g.fillRect(x + percent + 1, y + 1, percent_progress, bounds.height - 2);
```
Comment (conker-rsc) [inline src/Game/XPBar.java:161]: Actually I noticed the XP bar fill doesn't go all the way to the end of the bar if you're > lvl 1 on a skill. I think the below code fixes the problem, though please test edge conditions like lvl 99+

```suggestion
      g.fillRect(x + percent + 1, y + 1, percent_progress + (percent > 0.0f ? 1 : 0), bounds.height - 2);
```

Comment (lmsv-mx123) [reply src/Game/XPBar.java:162]: Good change but for it to look a bit more clean did that calculation when setting percent_progress

Comment (conker-rsc) [reply src/Game/XPBar.java:162]: good call
